### ISSUE TITLE
ipfs.dns does not handle subdomains correctly

### ISSUE THREAD

**Issue by niinpatel** [ipfs/js-ipfs#1932]
When I do ipfs.dns('docs.ipfs.io'). It throws an error 'ENODATA'. I did some digging and found out that this error is thrown by nodejs' `dns.resolveTxt` method when it does not find any txt records. 

Now, to handle subdomain correctly, we need to append `_dnslink.` string to it, which we do here
https://github.com/ipfs/js-ipfs/blob/3ef1ed42581152701d002efaaff4215c96f60930/src/core/runtime/dns-nodejs.js#L20-L22

But, the problem is, the error is way before the line gets to be executed, here: 

https://github.com/ipfs/js-ipfs/blob/3ef1ed42581152701d002efaaff4215c96f60930/src/core/runtime/dns-nodejs.js#L11

The easiest solution to this problem would be to check for err.code 'ENODATA' also along with 'ENOTFOUND', etc. in the if condition. So, the next lines get a chance to be executed. 

Basically something like this:

```
      if (err.code !== 'ENOTFOUND' && err.code !== 'ERR_DNSLINK_NOT_FOUND' && err.code !== 'ENODATA') throw err
```
Would fix the problem. 

What do you think?
